File: ObjectDock/Docklets/DotNet/Samples/DummyVB/Dummy.vb
Imports System
Imports System.Runtime.InteropServices
Imports System.Runtime.InteropServices.ComTypes
Imports System.Drawing
Imports System.Windows.Forms

Imports ObjectDockSDK
Imports ObjectDockSDK.Utils

Namespace DummyVB

    <Guid("81DE71EC-1635-4c7f-8675-EFDA8A5DCBE4")> _
    <ComVisible(True)> _
    <ClassInterface(ClassInterfaceType.None)> _
    Public Class DummyVB : Implements IDockletInterface

        Private Docklet As Docklet

        Public Sub OnGetInformation(ByRef name As String, ByRef author As String, ByRef version As Integer, ByRef notes As String) Implements IDockletInterface.OnGetInformation
            AssemblyData.GetInformation(name, author, version, notes)
        End Sub

        Public Sub OnCreate(ByVal data As DOCKLET_STATIC_DATA, ByVal ini As String, ByVal iniGroup As String) Implements IDockletInterface.OnCreate

            ODConsole.Initialize()

            Docklet = New Docklet(data)

            Docklet.Label = "DummyVB"
            Docklet.ImageFile = "DummyVB.png"

            Dim ContextMenu As Menu = New ContextMenu()

            Dim config As MenuItem = New MenuItem("Configure", AddressOf onMenuConfigure)

            Dim testmenu As MenuItem = New MenuItem("Test Menu")
            testmenu.MenuItems.Add("Item 1")
            testmenu.MenuItems.Add("Item 2")

            ContextMenu.MenuItems.Add(config)
            ContextMenu.MenuItems.Add(testmenu)

            Docklet.ContextMenu = ContextMenu
        End Sub

        Private Sub ShowInfo()
            'Label
            Dim sLabel As String = Docklet.Label
            'Visibility
            Dim sVisible As String = Docklet.IsVisible.ToString
            'Rectangle
            Dim sRect As String = Docklet.Rect.ToString
            'Edge & Align
            Dim sEdge As String = Docklet.DockEdge.ToString
            Dim sAlign As String = Docklet.DockAlign.ToString

            MsgBox(sLabel & vbCrLf & sVisible & vbCrLf & sRect & vbCrLf & sEdge & " - " & sAlign, MsgBoxStyle.Information)

        End Sub

        Public Sub OnSave(ByVal ini As String, ByVal iniGroup As String, ByVal isForExport As Boolean) Implements IDockletInterface.OnSave
            '
        End Sub

        Public Sub OnDestroy() Implements IDockletInterface.OnDestroy
            '
        End Sub

        Public Function OnExportFiles() As String() Implements IDockletInterface.OnExportFiles
            Dim oArray(2) As String

            oArray(0) = "DummyVB.png"
            oArray(1) = "DummyVB.dll"
            oArray(2) = "docklet.ini"

            Return oArray
        End Function

        Private Sub onMenuConfigure(ByVal sender As Object, ByVal e As EventArgs)
            Call OnConfigure()
        End Sub

        Public Function OnConfigure() As Boolean Implements IDockletInterface.OnConfigure
            Docklet.DoAttentionAnimation()
            Call ShowInfo()
            Docklet.DefaultConfigDialog()

            Return True
        End Function

        Public Function OnLeftButtonClick(ByRef ptCursor As Point, ByRef sizeDocklet As Size) As Boolean Implements IDockletInterface.OnLeftButtonClick
            Call OnConfigure()
            Return True
        End Function

        Public Function OnDoubleClick(ByRef ptCursor As Point, ByRef sizeDocklet As Size) As Boolean Implements IDockletInterface.OnDoubleClick
            Return False
        End Function

        Public Function OnLeftButtonHeld(ByRef ptCursor As Point, ByRef sizeDocklet As Size) As Boolean Implements IDockletInterface.OnLeftButtonHeld
            Return False
        End Function

        Public Function OnRightButtonClick(ByRef ptCursor As Point, ByRef sizeDocklet As Size) As Boolean Implements IDockletInterface.OnRightButtonClick
            Docklet.ShowContextMenu(ptCursor)
            Return True
        End Function

        Public Function OnAcceptDropFiles() As Boolean Implements IDockletInterface.OnAcceptDropFiles
            MsgBox("Accept Drop files...", MsgBoxStyle.Information)
            Return True
        End Function

        Public Sub OnDropFiles(ByVal hDrop As IntPtr) Implements IDockletInterface.OnDropFiles
            Dim sStr, sFiles As String
            Dim oArray() As String

            sStr = "Dropped file..." & vbCrLf & Docklet.GetCoordinates(hDrop).ToString() & vbCrLf

            oArray = Docklet.GetFiles(hDrop)

            sStr &= oArray.Length & vbCrLf

            For Each sFiles In oArray
                sStr &= sFiles & vbCrLf
            Next

            MsgBox(sStr, MsgBoxStyle.Information)

        End Sub

        Public Sub OnProcessMessage(ByVal hwnd As IntPtr, ByVal uMsg As UInteger, ByVal wParam As IntPtr, ByVal lParam As IntPtr) Implements IDockletInterface.OnProcessMessage

        End Sub

    End Class

End Namespace

File: trunk/ObjectDock/Docklets/DotNet/Samples/Dummy/Dummy.cs
///////////////////////////////////////////////////////////////////////////////////////////////
//
// Dummy .Net Docklet for ObjectDock
//
// Copyright (c) 2004-2008, Julien Templier
// All rights reserved.
//
///////////////////////////////////////////////////////////////////////////////////////////////
// * $LastChangedRevision$
// * $LastChangedDate$
// * $LastChangedBy$
///////////////////////////////////////////////////////////////////////////////////////////////
// 
// Redistribution and use in source and binary forms, with or without modification, are
// permitted provided that the following conditions are met:
//  1. Redistributions of source code must retain the above copyright notice, this list of
//     conditions and the following disclaimer. 
//  2. Redistributions in binary form must reproduce the above copyright notice, this list
//     of conditions and the following disclaimer in the documentation and/or other materials
//     provided with the distribution. 
//  3. The name of the author may not be used to endorse or promote products derived from this
//     software without specific prior written permission. 
//
//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS
//  OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
//  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
//  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
//  GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
//  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
//  OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
//  POSSIBILITY OF SUCH DAMAGE.
//
///////////////////////////////////////////////////////////////////////////////////////////////

using System;
using System.Runtime.InteropServices;
using System.Drawing;
using System.Windows.Forms;

using ObjectDockSDK;
using ObjectDockSDK.Utils;

namespace Dummy
{
	/// <summary>
	/// Dummy Docklet for ObjectDock (.NET)
	/// </summary>
	[Guid("847355FE-F9C3-4352-8FE2-FA7F68A9523D")]
	public class Dummy : IDockletInterface
	{
		private Docklet docklet;

		public void OnGetInformation(out string name, out string author, out int version, out string notes)
		{
			AssemblyData.GetInformation(out name, out author, out version, out notes);
		}

		public void OnCreate(DOCKLET_STATIC_DATA data, String ini, String iniGroup) 
		{
			ODConsole.Initialize();

			docklet = new Docklet(data);
	
			docklet.Label = "Dummy Docklet";
			docklet.ImageFile = "Dummy.png";

			#region Context Menu

			ContextMenu menu = new ContextMenu();
			MenuItem config = new MenuItem("Configure", new EventHandler(onMenuConfigure));
	
			MenuItem testmenu = new MenuItem("Test Menu");
            testmenu.MenuItems.Add("Browse For Image", new EventHandler(onBrowseForImage));
			testmenu.MenuItems.Add("Item 1");
			testmenu.MenuItems.Add("Item 2");

			menu.MenuItems.Add(config);
			menu.MenuItems.Add(testmenu);

			docklet.ContextMenu = menu;	

			#endregion
		}	    

	    private void ShowInfo()
		{
			// Label
			Console.WriteLine("Label: " + docklet.Label);
			
			// Visibility
			Console.WriteLine("Visibility: " + docklet.IsVisible + "\n");

			// Rectangle
			Console.WriteLine("Rectangle: " + docklet.Rect.ToString() + "\n");

			// Edge & Align
			Console.WriteLine("Edge: " + docklet.DockEdge + "\n");
			Console.WriteLine("Align: " + docklet.DockAlign + "\n");

		}

		public void OnSave(String ini, String iniGroup, Boolean isForExport) {	}

		public void OnDestroy()	{ }

		public String[] OnExportFiles()
		{
			return new String[] {"Dummy.png",
			 		     "Dummy.dll",
					     "docklet.ini"};
		}

		private void onMenuConfigure(object sender, EventArgs e)
		{
			OnConfigure();
		}

        private void onBrowseForImage(object sender, EventArgs e)
        {
            string image = "Dummy.png";
            bool ret = docklet.BrowseForImage(ref image, "");

            if (ret)
                Console.WriteLine("New Image: " + image + "\n");

        }

		public Boolean OnConfigure() 
		{
			docklet.DoAttentionAnimation();
			ShowInfo();
			docklet.DefaultConfigDialog();

			return true;
		}

		public Boolean OnLeftButtonClick(ref Point ptCursor, ref Size sizeDocklet)
		{
			OnConfigure();
			return true;
		}

		public Boolean OnDoubleClick(ref Point ptCursor, ref Size sizeDocklet)
		{
			return false;
		}

		public Boolean OnLeftButtonHeld(ref Point ptCursor, ref Size sizeDocklet)
		{
			return false;
		}

		public Boolean OnRightButtonClick(ref Point ptCursor, ref Size sizeDocklet)
		{
			docklet.ShowContextMenu(ptCursor);
			return true;
		}

		public Boolean OnAcceptDropFiles()
		{
			Console.WriteLine("Accept Drop files...");
			return true;
		}	

		public void OnDropFiles(IntPtr hDrop) {
			Console.WriteLine("Dropped file...");
		
			Console.WriteLine(docklet.GetCoordinates(hDrop).ToString());

			String[] files = docklet.GetFiles(hDrop);
			Console.WriteLine(files.Length);
			foreach (string str in files)
				Console.WriteLine(str);
		
		}

		public void OnProcessMessage(IntPtr hwnd, uint uMsg, IntPtr wParam, IntPtr lParam) {}
	}
}
